db/migrate/20190116182501_add_final_salvo_to_matches.rb

```diff
@@ -0,0 +1,7 @@
+class AddFinalSalvoToMatches < ActiveRecord::Migration[5.2]
+  def change
+    add_column :matches, :final_salvo, :integer
+    add_reference :matches, :final_salvo_winner, index: true
+    add_foreign_key :matches, :participants, column: :final_salvo_winner_id, primary_key: :id
```

Comment:
I don't think this is the quite the right approach. I think just having a winner_id column is good enough and then we won't have to worry about final salvo special cases or calculating the winner of non-final salvo matches.
This should also be added to the CreateMatches migration.

Reply:
Honestly I was just following the TTT format when I pulled that in. MetaWing doesn't take it into consideration and just calculates the winner on points or ignores it if it is a tie. I'll look at changing that last, because TTT and Cryodex use that format.